cmd/zc_traverser_list.go

```diff
@@ -78,6 +78,9 @@ func (l *listTraverser) traverse(processor objectProcessor, filters []objectFilt
 			return processor(object)
 		}
 
+		// The child traversers handle filters.
+		// Thus, --include-path xyz;abc --exclude-path def will actually exclude directory xyz/def and abc/def.
```

Comment:
In order to do this, we'd have to make slave traversers aware of meta-traversers handling filters.

Reply:
Can you double check that Adele?  I just added a test (to this PR) to try to reproduce what this code comment says, I  couldn't reproduce it.  (FWIW, I think the behaviour describe in the code comment is not desireable, so I think its _good_ that my test does not manifest the behaviour in the comment. - but need you to check my test, and think through what's going on, and amend this code comment if necessary).

Reply:
Really? Your test executes differently on my machine (and CI) in that case.

I understand the behavior being non-preferable. Generally speaking, I may've been thinking of the solution all wrong. As-is, we just pass all of our filters through to the sub-traversers. Hand them a nil list and then check in the decorators.

Reply:
Basically, include-path was being handled as a list traverser, which establishes traversers under new roots. The filters were being ran against the new roots. This comment is now incorrect due to me changing it so that no filters are given to the child traverser, and all filters occur in the decorators.

Reply:
I'm puzzled that the test passed for me (and for CI when I re-ran it just now here: https://github.com/Azure/azure-storage-azcopy/pull/610/checks?check_run_id=235162070 ). But I'll review the latest changes anyway (i.e. won't hold off reviewing just because of test mystery)

Reply:
Oh, I see. I had manipulated the test to make it pass with the current code!  It does indeed fail if I undo that manipulation of the test, and actually make it assert the behavior I think we should have. (I must have tweaked it to pass while we discussed the issue).